src/ui/actions/event-listeners.ts

```diff
@@ -66,13 +66,14 @@ export type NodeWithPreview = DeepRequired<
 >;
 
 export const isFunctionPreview = (obj?: ObjectPreview): obj is FunctionPreview => {
-  return (
-    !!obj && "functionName" in obj && "functionLocation" in obj && "functionParameterNames" in obj
-  );
+  return !!obj && "functionName" in obj && "functionLocation" in obj;
 };
 
 export const isFunctionWithPreview = (obj: ProtocolObject): obj is FunctionWithPreview => {
-  return obj.className === "Function" && isFunctionPreview(obj.preview);
+  return (
+    (obj.className === "Function" || obj.className === "AsyncFunction") &&
+    isFunctionPreview(obj.preview)
```

Comment:
I'm curious why we're checking this way (`"key" in obj`) rather than (`obj.key`)? Seems like this way would give false positives for empty strings or undefined/null values. Do we want that?

Reply:
The general intent is to check if the fields exist at all, and at that point we can assume that what's _in_ the fields matches the TS types from the backend.